app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java

```diff
@@ -0,0 +1,38 @@
+package pl.ppiwd.exerciseanalyst.common.auth;
+
+public final class TokenStore {
+    private static final Object lock = new Object();
+    private static TokenStore instance = null;
+
+    private String accessToken;
+
+    public static TokenStore getInstance() {
+        TokenStore tokenStore = instance;
+        if (tokenStore == null) {
+            synchronized (lock) {    // While we were waiting for the lock, another
+                tokenStore = instance;        // thread may have instantiated the object.
+                if (tokenStore == null) {
+                    tokenStore = new TokenStore();
+                    instance = tokenStore;
+                }
+            }
+        }
+        return tokenStore;
+    }
+
+    private TokenStore() {
+        wipeToken();
+    }
+
+    public void storeToken(final String accessToken) {
```

Comment:
assuming that it should be thread-safe then this has to be synchronized

Reply:
You are right!